devito/data/decomposition.py

```diff
@@ -241,21 +259,49 @@ def convert_index(self, *args, rel=True):
                 if glb_idx_max is not None and glb_idx_max < 0:
                     glb_idx_max = self.glb_max + glb_idx_max + 1
                 # -> Do the actual conversion
-                if glb_idx_min is None or glb_idx_min < self.loc_abs_min:
+                if isinstance(glb_idx, slice) and glb_idx.step is not None \
+                        and glb_idx.step > 1:
+                    if glb_idx_min > self.loc_abs_max:
+                        return retfunc(-1, -3)
+                    elif glb_idx.start is None:
+                        loc_min = self.loc_abs_min - base \
+                            + np.mod(glb_idx.step - np.mod(base, glb_idx.step),
+                                     glb_idx.step)
+                    else:
+                        loc_min = self.loc_abs_min - base \
+                            + np.mod(glb_idx.step - np.mod(base - glb_idx.start,
+                                                           glb_idx.step), glb_idx.step)
+                elif glb_idx_min is None or glb_idx_min < self.loc_abs_min:
                     loc_min = self.loc_abs_min - base
                 elif glb_idx_min > self.loc_abs_max:
                     return retfunc(-1, -3)
                 else:
                     loc_min = glb_idx_min - base
-                if glb_idx_max is None or glb_idx_max > self.loc_abs_max:
+                if isinstance(glb_idx, slice) and glb_idx.step is not None \
+                        and glb_idx.step < -1:
+                    if glb_idx_max < self.loc_abs_min:
+                        return retfunc(-1, -3)
+                    elif glb_idx.start is None:
```

Comment:
I could use a comment here? or maybe assign ``np.mod(....)`` to a variable with a name such that I can vaguely grasp what's going on?

Reply:
Comments added.